client/src/app/components/CreateProject/Form.js

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import ButtonContainer from "../Button/Button";
+import "./styles.css";
+
+class Form extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      titleValue: "",
+      descriptionValue: "",
+      languages: [],
+      complexity: "",
+      members: 0,
+      reset: false
+    };
+  }
+
+  onUpdateSelection(buttonTitle, type) {
+    if (type === "language") {
+      this.setState(prevState => ({
+        languages: [...prevState.languages, buttonTitle]
+      }));
+    } else if (type === "complexity") {
+      this.setState({ complexity: buttonTitle });
+    } else if (type === "members") {
+      this.setState({ members: buttonTitle });
+    }
+  }
+
+  onChangeInput(e) {
+    e.target.name === "project-title"
+      ? this.setState({ titleValue: e.target.value })
+      : this.setState({ descriptionValue: e.target.value });
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    console.log(this.state);
+    this.setState(
+      () => ({
+        title: "",
+        description: "",
+        languages: [],
+        complexity: "",
+        members: 0,
+        reset: true
+      }),
+      () => {
+        this.setState({ reset: false });
+      }
+    );
+  }
+
+  render() {
+    return (
```

Comment:
It makes the code cleaner to destructure any props and state on top of the render. At least 2 advantages: 1. It makes it much clearer what your 'render' is using, and of course, 2. the code will look cleaner without this.props.sth and this.state.sth repeated each time